elastica/memory_block/memory_block_rod.py

```diff
@@ -33,7 +31,7 @@ class MemoryBlockCosseratRod(
     """
 
     def __init__(
-        self, systems: list[CosseratRod], system_idx_list: list[SystemIdxType]
+        self, systems: list[RodBase], system_idx_list: list[SystemIdxType]
```

Comment:
Difference in the constitutive model is okay, as long as muscle rod is derived from RodBase.

This update does not contain full type-hinting for rod. It will be included in #396.